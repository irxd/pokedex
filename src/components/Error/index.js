import React from 'react';
import Snackbar from '@material-ui/core/Snackbar';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

function Error(props) {
  const { error, handleCloseSnackbar } = props;
  
  return (
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
      open={error}
      autoHideDuration={2000}
      onClose={handleCloseSnackbar}
      message="Not found"
      action={
        <React.Fragment>
          <IconButton size="small" aria-label="close" color="inherit" onClick={handleCloseSnackbar}>
            <CloseIcon fontSize="small" />
          </IconButton>
        </React.Fragment>
      }
    />
  );
}

export default Error;
