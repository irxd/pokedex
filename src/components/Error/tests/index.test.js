import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Error from '../index';

const renderer = new ShallowRenderer();

describe('<Error />', () => {
  it('Render error component and should match the snapshot', () => {
    renderer.render(
      <Error
        error={true}
        handleCloseSnackbar={() => {}}
      />
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
