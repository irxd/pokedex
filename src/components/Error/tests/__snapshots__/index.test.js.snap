// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> Render error component and should match the snapshot 1`] = `
<ForwardRef(WithStyles)
  action={
    <React.Fragment>
      <ForwardRef(WithStyles)
        aria-label="close"
        color="inherit"
        onClick={[Function]}
        size="small"
      >
        <UNDEFINED
          fontSize="small"
        />
      </ForwardRef(WithStyles)>
    </React.Fragment>
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={2000}
  message="Not found"
  onClose={[Function]}
  open={true}
/>
`;
